sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -129,12 +129,16 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
         conditionSQL = condition.sql
       } yield s"$childSQL $whereOrHaving $conditionSQL"
 
-    case Union(left, right) =>
+    case Union(children) if children.length > 1 =>
```

Comment:
`SQLBuilder` is run for analyzed plan, so it's guaranteed `Union` has more than 1 children right?